include/MRFactory.py

```diff
@@ -0,0 +1,215 @@
+	def __init__(self, geometry, properties, osmType, osmId):
+		super().__init__(geometry, properties)
+		self.osmId = osmId
+		self.osmType = osmType
+
+	def to_dict(self):
+		return {
+			"type": "Feature",
+			"geometry": {
+				"type": self.geometryType,
+				"coordinates": self.geometry
+			},
+			"properties": {
+				"@id": str(self.osmType) + "/" + str(self.osmId),
+				**self.properties
+			}
+		}
+
+@dataclass
+class TagFix():
+	def __init__(self, osmType, osmId, tags):
+		# give tags to set and to delete in this format: {"name": "suggested name", "wrongtag": null}
+		self.osmType = osmType
+		self.osmId = osmId
+		self.tagsToDelete = [key for key, value in tags.items() if value == None]
+		self.tagsToSet = {key: value for key, value in tags.items() if value != None}
+		if not isinstance(self.tagsToDelete, list):
+			raise ValueError("tagsToDelete must be a list, e.g. ['tag1', 'tag2']")
+		if not isinstance(self.tagsToSet, dict):
+			raise ValueError("tagsToSet must be a dict e.g. {'tag1': 'value1', 'tag2': 'value2'}")
+		
+	def to_dict(self):
+		return {"meta": 
+			{"version": 2, "type": 1}, 
+			"operations": [
+				{"operationType": "modifyElement", 
+				"data": {
+					"id": str(self.osmType) + "/" + str(self.osmId),  
+					"operations": [
+						{
+							"operation": "setTags", 
+							"data": self.tagsToSet
+						},
+						{
+							"operation": "unsetTags",
+							"data": self.tagsToDelete
+						}
+					]
+				}}]}
+
+@dataclass
+class Task:
+	def __init__(self, mainFeature, additionalFeatures = [], cooperativeWork = None):
+		self.mainFeature = mainFeature
+		self.additionalFeatures = additionalFeatures
+		self.cooperativeWork = cooperativeWork
+
+	def to_dict(self):
+		# the features are the main feature and the additional features as one list
+		features = [self.mainFeature.to_dict()]
+		for feature in self.additionalFeatures:
+			features.append(feature.to_dict())
+		if self.cooperativeWork == None:
+			return {
+				"type": "FeatureCollection",
+				"features": features,
+			}
+		else:
+			return {
+				"type": "FeatureCollection",
+				"features": features,
+				"cooperativeWork": self.cooperativeWork.to_dict()
+			}
+
+@dataclass
+class Challenge:
+	def __init__(self):
+		self.tasks = []
+
+	def addTask(self, task):
+		self.tasks.append(task)
+
+	def saveToFile(self, filename):
+		with open(filename, 'w', encoding="UTF-8") as f:
+			for task in self.tasks:
+				f.write('\x1E')
+				json.dump(task.to_dict(), f, ensure_ascii=False)
+				f.write('\n')
+
+class Overpass:
+	def __init__(self):
+		self.resultElements = {}
+		self.overpass_url = "https://overpass-api.de/api/interpreter"
+
+	def getElementsFromQuery(self, overpass_query):
+		response = requests.get(self.overpass_url, params={'data': overpass_query})
+		if response.status_code != 200:
+			raise ValueError("Invalid return data")
+		resultElements = response.json()["elements"]
+		return resultElements
+
+
+def createElementCenterPoint(element):
```

Comment:
I still believe it is worth thinking about a "Geometry" class. The idea is to create the true geometry and then allow to get simplifications of it via functions like "center" oder "bbox". Again for the bbox and center methods I recommend using a geo library.

```
class Geometry {
   constructor fromElement();
   center();
   bbox();
}
```